Update Azure OIDC steps (#1885)
astronomer/docs: software/integrate-auth-system.md

> Replace the values above with those of the provider of your choice. If you want to configure Azure AD, Okta or Auth0 read below for specific guidelines.

Reviewer: ```suggestion
Replace the values in the example with your provider values. To configure Azure AD, Okta, or Auth0, refer to one of the following sections.
```
![image](https://user-images.githubusercontent.com/104205257/218544161-dae88ad6-76e8-475c-9ee2-1ef79491432a.png)

### AWS Cognito

#### Create a user pool in Cognito

> Start by creating a user pool in Cognito. You can either review the default settings or step through them to customize.

Reviewer: This introductory text should be replaced with text that provides more background. Why is the user creating a user pool in Cognito? Why is it required?

Author: As you can see from the file changes and the topic of the PR, I'm not editing these topics beyond fixing the header levels. Resolving these comments now to focus on the core change in the PR

Reviewer: @jwitz why not take the time to implement the changes? Shouldn't we be making incremental improvements whenever we can? I've provided the edit, all you need to do is commit it.

Author: I get what you're saying for the most part and implemented your grammar changes to other unedited sections! But because this PR is about Azure AD, I'm not going to be able to answer the "why" of a tool that I haven't touched since I joined Astronomer.

> This should give you a minimally working user pool configuration.

Reviewer: ```suggestion
This should provide a functional user pool configuration.
```
Is this necessary?

> 
> This should give you a minimally working user pool configuration.

Author: ```suggestion
```

Your Cognito pool ID can be found in the `General settings` tab of the Cognito portal. Your client ID is found in the `App clients` tab.

Once you've saved your `config.yaml` file with these values, push it to your platform. See [Apply a config change](apply-platform-config.md).

### Azure AD

Follow these steps to configure Azure AD as your OIDC provider. 

#### Register the Application using `App Registrations` on Azure

1. In Azure Active Directory, click **App registrations** > **New registration**. 

> 2.Complete the following sections:

Author: ```suggestion
2. Complete the following sections:
```

#### Register the Application using `App Registrations` on Azure

1. In Azure Active Directory, click **App registrations** > **New registration**. 
2.Complete the following sections:
  
    - **Name**: Any
    - **Supported account types**: Accounts in this organizational directory only (Astronomer only - single tenant)
    - **Redirect URI**: Web / `https://houston.BASEDOMAIN/v1/oauth/redirect/`.

    Replace `BASEDOMAIN` with your own. For example, if your base domain is `mycompany.com`, your redirect URI is https://houston.mycompany.com/v1/oauth/redirect/

3. Click **Register**.

4. Click **Authentication** in the left menu.
5. In the **Web** area, confirm the redirect URI is correct.

6. In the **Implicit grant and hybrid flows** area, select **Access tokens** and **ID tokens**.

7. Click **Save**.

![authentication.png](/img/software/azure-authentication.png)

#### Create a client secret

1. In your Azure AD application management left menu, click **Certificates & secrets**.
2. Click **New client secret**.
3. Enter a description in the **Description** field and then select an expiry period in the **Expires** list. 

> 
> 4. Click **Add**.
> 

Author: ```suggestion
4. Click **Add**.
```

5. Copy the values in the **Value** and **Secret ID** columns. 
6. Click **API permissions** in the left menu.

> 5. At a minimum, grant your application the following permissions:

Author: ```suggestion
5. Click **Microsoft Graph** and add the following minimum permissions for Microsoft Graph:
```

> Once you've saved your `config.yaml` file with these values, push it to your platform as described in [Apply a config change](apply-platform-config.md).

Reviewer: ```suggestion
Once you've saved your `config.yaml` file with these values, push it to your platform. See [Apply a config change](apply-platform-config.md).
```

> ### Azure AD

Reviewer: ```suggestion
### Azure AD
```
Add an introductory statement below the heading.

![image](https://user-images.githubusercontent.com/104205257/218545407-ca661925-565c-4d1f-9451-d7f58f1a7644.png)

#### Register the Application using `App Registrations` on Azure

> 1. In the Azure portal, click **New registration**. 

Reviewer: ```suggestion
1. In Azure Active Directory, click **App registrations** > **New registration**. 
```
![image](https://user-images.githubusercontent.com/104205257/218546921-5f2c81f4-bad7-469d-81a9-4f8185fe9d9d.png)

> 2. Configure the application as follows:

Reviewer: ```suggestion
2.Complete the following sections:
```

  
    - **Name**: Any
    - **Supported account types**: Accounts in this organizational directory only (Astronomer only - single tenant)
    - **Redirect URI**: Web / `https://houston.BASEDOMAIN/v1/oauth/redirect/`.

>     Replace `BASEDOMAIN` with your own. For example, if your base domain were `mycompany.com`, your redirect URI would be https://houston.mycompany.com/v1/oauth/redirect/

Reviewer: ```suggestion
    Replace `BASEDOMAIN` with your own. For example, if your base domain is `mycompany.com`, your redirect URI is https://houston.mycompany.com/v1/oauth/redirect/
```

> 3. In your Azure AD application management left menu, click **Authentication**.

Reviewer: ```suggestion
3. Click **Register**.

4. Click **Authentication** in the left menu.
```

> 4. Ensure that Access Tokens and ID Tokens are enabled, as well as to verify the redirect URI.

Reviewer: ```suggestion
5. In the **Web** area, confirm the redirect URI is correct.

6. In the **Implicit grant and hybrid flows** area, select **Access tokens** and **ID tokens**.

7. Click **Save**.
```
![image](https://user-images.githubusercontent.com/104205257/218548828-87e8e83a-6ad4-4aac-83c8-b4315230fbc9.png)

![image](https://user-images.githubusercontent.com/104205257/218548668-4b276ac5-e42f-4ac1-ab2c-53d17c36ade2.png)

![authentication.png](/img/software/azure-authentication.png)

#### Create a client secret

1. In your Azure AD application management left menu, click **Certificates & secrets**.
2. Click **New client secret**.

> 3. Create a new client secret. Copy both the client ID and secret. 

Reviewer: ```suggestion
3. Enter a description in the **Description** field and then select an expiry period in the **Expires** list. 

4. Click **Add**.

5. Copy the values in the **Value** and **Secret ID** columns. 
```
At this point, there isn't a client ID and secret.

![image](https://user-images.githubusercontent.com/104205257/218550584-92dfad37-1e15-40ea-97c1-e48954696c68.png)

> 4. In your Azure AD application management left menu, click **API permissions**.

Reviewer: ```suggestion
6. Click **API permissions** in the left menu.
```

5. At a minimum, grant your application the following permissions:

>     - `email`

Reviewer: These are not the permissions in the Azure UI.  There are three tabs:

![image](https://user-images.githubusercontent.com/104205257/218553606-1f7c6020-f225-4ed4-af5a-39942f7d9e4b.png)

This requires some clarification.

Author: Updated to explain how to access the permissions based on my example App expierience
![Screenshot 2023-02-14 at 2 03 01 PM](https://user-images.githubusercontent.com/74574233/218761441-eca19bc8-2d41-464f-b11c-c61915062d9a.png)

    - `Group.Read.All`
    - `openid`
    - `profile`
    - `User.Read`

5. In your Azure AD application management left menu, click **Token configuration**.
6. Click **Add groups claim**. And configure a groups claim as follows:

>     - In **Select group types to include in Access, ID, and SAML tokens**, check every check box. 

Reviewer: ```suggestion
    - Select every option in the **Select group types to include in Access, ID, and SAML tokens** area. 
```

>     - In **Customize token properties by type**, select **Group ID** for all types.

Reviewer: ```suggestion
    - In **Customize token properties by type** area, expand **ID**, **Access**, and **SAML** and then select **Group ID** for each type.
```

#### Enable Azure AD in your config.yaml file

> Make sure the `config.yaml` file in your `astronomer` directory is updated with the proper values:

Reviewer: ```suggestion
Add the following values to the `config.yaml` file in your `astronomer` directory:
```

            enabled: false
          microsoft:
            enabled: true
            clientId: <your-client-id>
            discoveryUrl: https://login.microsoftonline.com/<tenant-id>/v2.0/
            clientSecret: <your-client-secret>
            authUrlParams:
              audience: <cyour-client-id>
        github:
          enabled: false
```

> Then, push the configuration change to your platform as described in [Apply a config change](apply-platform-config.md).

Reviewer: ```suggestion
Push the configuration change to your platform. See [Apply a config change](apply-platform-config.md).
```

### Okta

> To integrate Okta with Astronomer, you'll need to make configuration changes both within Okta and on Astronomer.

Reviewer: ```suggestion
To integrate Okta with Astronomer, you'll need to make configuration changes in Okta and Astronomer.
```

> >> **Note:** `okta-base-domain` will be different from the basedomain of your Software installation. You can read [Okta's docs on finding your domain](https://developer.okta.com/docs/api/getting_started/finding_your_domain/) if you are unsure what this value should be.

Reviewer: ```suggestion
>> **Note:** `okta-base-domain` will be different from the basedomain of your Software installation. You can read [Okta's docs on finding your domain](https://developer.okta.com/docs/api/getting_started/finding_your_domain/) if you are unsure what this value should be.
```
Should this be updated to one of our annotation styles?

> >> **Note:** `okta-base-domain` will be different from the basedomain of your Software installation. You can read [Okta's docs on finding your domain](https://developer.okta.com/docs/api/getting_started/finding_your_domain/) if you are unsure what this value should be.

Author: ```suggestion
> **Note:** `okta-base-domain` will be different from the basedomain of your Software installation. You can read [Okta's docs on finding your domain](https://developer.okta.com/docs/api/getting_started/finding_your_domain/) if you are unsure what this value should be.
```
